front-end-tools/src/utils.ts

```diff
@@ -20,3 +29,151 @@ export function getObjectExample(template: string | undefined) {
+                break;
+            case -2147483636:
+                humanReadableError = '`[NotPayableError]`';
+                break;
+            case -2147483635:
+                humanReadableError = '`[TransferError::AmountTooLarge]`';
+                break;
+            case -2147483634:
+                humanReadableError = '`[TransferError::MissingAccount]`';
+                break;
+            case -2147483633:
+                humanReadableError = '`[CallContractError::AmountTooLarge]`';
+                break;
+            case -2147483632:
+                humanReadableError = '`[CallContractError::MissingAccount]`';
+                break;
+            case -2147483631:
+                humanReadableError = '`[CallContractError::MissingContract]`';
+                break;
+            case -2147483630:
+                humanReadableError = '`[CallContractError::MissingEntrypoint]`';
+                break;
+            case -2147483629:
+                humanReadableError = '`[CallContractError::MessageFailed]`';
+                break;
+            case -2147483628:
+                humanReadableError = '`[CallContractError::LogicReject]`';
+                break;
+            case -2147483627:
+                humanReadableError = '`[CallContractError::Trap]`';
+                break;
+            case -2147483626:
+                humanReadableError = '`[UpgradeError::MissingModule]`';
+                break;
+            case -2147483625:
+                humanReadableError = '`[UpgradeError::MissingContract]`';
+                break;
+            case -2147483624:
+                humanReadableError = '`[UpgradeError::UnsupportedModuleVersion]`';
+                break;
+            case -2147483623:
+                humanReadableError = '`[QueryAccountBalanceError]`';
+                break;
+            case -2147483622:
+                humanReadableError = '`[QueryContractBalanceError]`';
+                break;
+            default:
+                // If the `rejectReason` comes from the smart contract itself (e.g. -1, -2, -3, ...)
+                // and an error schema is provided in the module schema, deserialize the reject reason into a human-readable string.
+                if (moduleSchema !== undefined && failedResult.returnValue !== undefined) {
+                    const decodedError = deserializeReceiveError(
+                        ReturnValue.toBuffer(failedResult.returnValue),
+                        toBuffer(moduleSchema, 'base64'),
+                        contractName,
+                        entryPoint
+                    );
+
+                    // The object only includes one key. Its key is the human-readable error string.
+                    const key = Object.keys(decodedError);
+
+                    // Convert the human-readable error to a JSON string.
+                    humanReadableError = JSON.stringify(key);
+                }
```

Comment:
So why are we not doing this always?

I'm looking at wccd for example. It will have some error codes like in the list above, but for example
```

            case -2147483645:
                humanReadableError = '`[LogError::Full]`';
                break;
            case -2147483644:
                humanReadableError = '`[LogError::Malformed]`';
                break;
```
these will not be matched since we `derive(Reject)` for them.

So I'm not quite sure I see the reason for trying to map reject reasons first. The schema based serialization will be better as far as I see it.

I could see a reason for having a fallback to reject reason parsing if the schema is not there.